File: data/readable_format/fortran/file110.f90
module chtest
contains
  function chararray2string(chararray) result(text)
    character(len=1), dimension(:) :: chararray
    character(len=size(chararray)) :: text
    do i = 1,size(chararray)
      text(i:i) = chararray(i)
    end do
  end function chararray2string

  subroutine testChararray2string()
    character(len=1), dimension(2) :: testArray
    character(len=2) :: resultString
    
    ! Test case 1: Convert "AB" to a string
    testArray = ['A', 'B']
    resultString = chararray2string(testArray)
    if (resultString .ne. "AB") then
      print *, "Test Failed: AB"
    else
      print *, "Test Passed: AB"
    endif
  end subroutine testChararray2string
end module chtest

program TestStringTools
  use chtest
  call testChararray2string()
end program TestStringTools

File: data/readable_format/cpp/file110.cpp
#include <gtest/gtest.h>
#include <string>
#include <vector>

// Function to convert a vector of characters to a string
std::string charArray2String(const std::vector<char>& charArray) {
    std::string text;
    for (const char& c : charArray) {
        text += c;
    }
    return text;
}

// Test for charArray2String function
TEST(CharArray2StringTest, ConvertsCorrectly) {
    std::vector<char> testArray = {'A', 'B', 'C', 'D'};
    std::string result = charArray2String(testArray);
    ASSERT_EQ(result, "ABCD");
}

// Test for the entire program logic
TEST(ProgramTest, GeneratesCorrectString) {
    // Generate the character array
    std::vector<char> charArr;
    for (int i = 1; i <= 26; ++i) {
        charArr.push_back(char(i + 64)); // Uppercase letters
        charArr.push_back(char(i + 96)); // Lowercase letters
    }

    // Convert to string and compare
    std::string txt = charArray2String(charArr);
    ASSERT_EQ(txt, "AaBbCcDdEeFfGgHhIiJjKkLlMmNnOoPpQqRrSsTtUuVvWwXxYyZz");
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}